atomic_reactor/inner.py

```diff
@@ -404,6 +596,168 @@ def _update(data):
         return new_data
 
 
+@dataclass
+class ImageBuildWorkflowData(ISerializer):
+    """Manage workflow data.
+
+    Workflow data is those data which are generated values by plugins through
+    the whole build workflow (pipeline) and must be shared in every build tasks.
+
+    These data can be dumped into dictionary object in order to be saved into a
+    file as JSON data, and then be loaded while atomic-reactor launches again to
+    execute another set of plugins for a different build task.
+    """
+
+    dockerfile_images: DockerfileImages = field(default_factory=DockerfileImages)
+    tag_conf: TagConf = field(default_factory=TagConf)
+    push_conf: PushConf = field(default_factory=PushConf)
+
+    prebuild_results: Dict[str, Any] = field(default_factory=dict)
+    buildstep_result: Dict[str, Any] = field(default_factory=dict)
+    postbuild_results: Dict[str, Any] = field(default_factory=dict)
+    prepub_results: Dict[str, Any] = field(default_factory=dict)
+    exit_results: Dict[str, Any] = field(default_factory=dict)
+    plugin_workspace: Dict[str, Any] = field(default_factory=dict)
+
+    # Plugin name -> timestamp in isoformat
+    plugins_timestamps: Dict[str, str] = field(default_factory=dict)
+    # Plugin name -> seconds
+    plugins_durations: Dict[str, float] = field(default_factory=dict)
+    # Plugin name -> a string containing error message
+    plugins_errors: Dict[str, str] = field(default_factory=dict)
+    build_canceled: bool = False
+    plugin_failed: bool = False
+
+    # info about pre-declared build, build-id and token
+    reserved_build_id: int = None
+    reserved_token: str = None
+    koji_source_nvr: Dict[str, str] = field(default_factory=dict)
+    koji_source_source_url: str = None
+    koji_source_manifest: Dict[str, Union[int, str, ManifestInfo, List[ManifestInfo]]] = None
+
+    buildargs: Dict[str, str] = field(default_factory=dict)  # --buildargs for container build
+    built_image_inspect: Dict[str, Any] = None
+    layer_sizes: List[Dict[str, Union[str, int]]] = field(default_factory=list)
+
+    # list of images pulled during the build, to be deleted after the build
+    pulled_base_images: Set = field(default_factory=set)
+
+    # When an image is exported into tarball, it can then be processed by various plugins.
+    #  Each plugin that transforms the image should save it as a new file and append it to
+    #  the end of exported_image_sequence. Other plugins should then operate with last
+    #  member of this structure. Example:
+    #  [{'path': '/tmp/foo.tar', 'size': 12345678, 'md5sum': '<md5>', 'sha256sum': '<sha256>'}]
+    #  You can use util.get_exported_image_metadata to create a dict to append to this list.
+    exported_image_sequence: List[Dict[str, Union[str, int]]] = field(default_factory=list)
+
+    # mapping of downloaded files; DON'T PUT ANYTHING BIG HERE!
+    # "path/to/file" -> "content"
+    files: Dict[str, str] = field(default_factory=dict)
```

Comment:
we don't

Reply:
`files` is still used in `add_yum_repo_by_url` currently. The basic idea of making this PR for manage workflow data is that, part of the work is just to refactor the data attributes from DockerBuildWorkflow to a specific workflow data class. Since part of the attributes, that are planned to be removed, are still being used in plugins, they are not removed from the workflow data in this PR to avoid making the scope bigger.

So, similarly with the plugins_workspace to the flatpak plugins, when the implementation of writing repos into the context_dir as files is done, `files` can be removed from the workflow data.

This consideration also applies to the registry classes mentioned above that will be replaced with arch-specific registries.